### DISCUSSION TITLE
Is there any example that can make sprite move on a specified color background

### DISCUSSION THREAD

**Discussion by caesar-empereur** [Bodmer/TFT_eSPI#3206] [Q&A - Sprites]
Greetings

I wish to use draw several elements (like filled circles) floating/moving on the background image (without messing the background) I tried to use TFT_eSprite as the foreground and push them, but noticed it leaves a trace on the background

**Comment by Bodmer** [comment]
The example here does that:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Sprite/Sprite_draw

**Reply by caesar-empereur** [reply]
Thanks!
I found the code like this:

void setup(){
  tft.init();
  spr.createSprite(WIDTH, HEIGHT);
  tft.fillScreen(TFT_BLUE);
}
  tft.fillScreen(TFT_BLUE);

  // **Draw a blue rectangle in sprite so when we move it 1 pixel it does not leave a trail**
  // on the blue screen background
  spr.drawRect(0, 0, WIDTH, HEIGHT, TFT_BLUE);

But the prerequisite for this example to be effective is that you know what color the background of the entire screen is, and it is a single color. In my example, the background of the screen is composed of multiple colors spliced together, and the area where my sprite is located may contain several colors. How should I handle this
